docs/devel/consumers-claims.md

```diff
@@ -0,0 +1,309 @@
+scheduled with a claim that is not yet bound, the kubelet will refuse to start
+the `Pod` until the claim is bound.  In the future this should be handled
+before it ever hits the kubelet.
+
+### Config
+
+We expect that `VolumeManager`s will need some amount of configuration, and it
+will be convenient if they have similar parameters, but this should be by
+convention, not by diktat.  Example parameters:
+   * Max instances - limit of number of volumes allowed
+   * Min free instances - how many ready-to-use volumes should be maintained
+   * Max free instances - limit of number of ready-to-use volumes
+   * Size distribution - desired histogram of sizes of ready-to-use volumes
+
+In addition to these, there will be volume-type specific parameters, such as
+the GCE SSD vs Disk param described above.
+
+### Scrubbing
+
+Volumes need to be scrubbed before they can be reused, and some storage devices
+may even need to be scrubbed before they are decommissioned.  This is the
+responsibility of the `VolumeManager` as it drives the lifecycle.
+
+## Networking
+
+TODO: flip Ingress and LB ?
+
+Network ingress in Kubernetes will be represented by `LoadBalancer`s.  A user
+can create an `IngressClaim` and reference that claim from an `Ingress` object.
+A `LoadBalancerManager` is responsible for binding `IngressClaim`s to
+`LoadBalancers`s.  If the `IngressClaim` can not be satisified by any existing
+`LoadBalancer`, the `LoadBalancerManager` may provision a new one.
+
+```
+                       +--------------+
+        +--create----> | LoadBalancer |
+        |              +--------------+
+   +----+----+               ^
+   |  LBMgr  |---bind------> :
+   +----+----+               v
+        |              +---------------+
+        +---watch----> | IngressClaim  | <--+
+                       +---------------+    |
+                                 ^          |
+                                 |          +-------- user-created
+                                 |          |
+                           +----------+     |
+                           | Ingress  | <---+
+                           +----------+
+```
+
+### IngressClaim
+
+An `IngressClaim` does not specify which specific Ingress it wants.  Instead it
+specifies characteristics of the ingress it needs and those characteristics are
+satisfied as best is possible.  Characteristics might include the availability
+properties required or the specific IP address to use.
+
+Like `PersistentVolumeClaim`, users will be able to request coarse "classes" of
+ingress (e.g. Gold, Silver, Bronze).  How those classes map to real storage
+technologies is a policy set by the cluster administrators.
+
+### LoadBalancerManagers
+
+The act of binding a `IngressClaim` to an `Ingress` is the responsibility of a
+process in the cluster.  A cluster may have any number of
+`LoadBalancerManager`s running, each of which handles some set of claim
```

Comment:
I always find it confusing when components are referred to using code-style syntax. Do we write `Scheduler`? We don't, or at least we shouldn't. Use ordinary English for components and code-style syntax for API types.